lib/rules/sort-imports.js

```diff
@@ -83,62 +154,139 @@ module.exports = {
         }
 
         /**
-         * Gets the local name of the first imported module.
-         * @param {ASTNode} node - the ImportDeclaration node.
-         * @returns {?string} the local name of the first imported module.
+         * Sorts import statements
+         *
+         * @param {nodeA} nodeA ImportDeclaration node.
+         * @param {nodeB} nodeB  ImportDeclaration node.
+         * @returns {number} returns a number - result of sort
          */
-        function getFirstLocalMemberName(node) {
-            if (node.specifiers[0]) {
-                return node.specifiers[0].local.name;
-            } else {
-                return null;
+        function sortingFunction(nodeA, nodeB) {
+            const nodeASource = nodeA.source.value;
+            const nodeBSource = nodeB.source.value;
+
+            // react imports always goes first
+            if (reactRegExp.test(nodeASource) && reactRegExp.test(nodeBSource)) {
+                if (nodeASource < nodeBSource) {
+                    return -1;
+                }
+
+                if (nodeASource > nodeBSource) {
+                    return 1;
+                }
+            } else if (reactRegExp.test(nodeASource)) {
```

Comment:
The result of `reactRegExp.test(nodeASource))` (and same for `nodeBSource`) should already be known. Consider storing the results in a variable, instead of doing the same regex operation twice.